### DISCUSSION TITLE
`UIManagerModule.addUIBlock` not working in New Architecture

### DISCUSSION THREAD

**Discussion by joemun** [reactwg/react-native-new-architecture#148] [Q&A] [answered]
Someone had filed this issue over in the RN repo here: https://github.com/facebook/react-native/issues/34764
I'm picking up where that left off, as I'm also experiencing this issue.

It looks like [UIManagerModule.addUIBlock()](https://github.com/facebook/react-native/blob/4d9bd9040107661e336cddad1948f50b78ce8ad6/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/UIManagerModule.java#L753-L766) doesn't work when New Arch is enabled.  The callback is not called.

Is this expected?  For New Architecture, was `addUIBlock()` replaced with another API? (I did find `UiThreadUtil.runOnUiThread()`, but I'm not sure if that's meant to be a suitable replacement...)

I've created a repro based on the RN Reproducer app here: https://github.com/joemun/uimanagermodule-adduiblock-test
The client calls the Turbo Module function, which is then supposed to resolve the promise in the `addUIBlock()` callback, but the callback is not called.   If `newArchEnabled` is then set to false, the callback is called.

**Comment by arushikesarwani94** [comment, marked as answer]
Apologies for the delayed response. Yes, that's correct for New Arch [UIManagerModule.addUIBlock()](https://github.com/facebook/react-native/blob/4d9bd9040107661e336cddad1948f50b78ce8ad6/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/UIManagerModule.java#L753-L766) returns [null](https://github.com/facebook/react-native/blob/main/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/runtime/ReactHostImpl.java#L600-L606) on Bridgeless.

The replacement API added for Backwards Compatibility in Bridgeless is 
https://github.com/facebook/react-native/blob/2483c63017b0d47c377bf67cb7c597e2d7e57897/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/fabric/FabricUIManager.java#L104

Sample fix PRs : https://github.com/expo/expo/pull/29500 https://github.com/react-native-maps/react-native-maps/pull/5061

**Reply by hwhh** [reply]
Do you have an example for iOS swift, I've been trying to migrate [this file](https://github.com/Fouppy/react-native-money/blob/main/ios/TextInputMask.swift) for the last couple of days but am struggling: 

any help would be greatly greatly appreciated!

```
var methodQueue: DispatchQueue {
        bridge.uiManager.methodQueue
    }

 var bridge: RCTBridge!
    
    
@objc(initializeMoneyInput:options:)
    func initializeMoneyInput(reactNode: NSNumber, options: NSDictionary) {
        bridge.uiManager.addUIBlock { (uiManager, viewRegistry) in
            DispatchQueue.main.async {
                guard let view = viewRegistry?[reactNode] as? RCTBaseTextInputView else { return }
                let textView = view.backedTextInputView as! RCTUITextField
            
                let locale = options["locale"] as? String
                let maskedDelegate = MoneyInputDelegate(localeIdentifier: locale) { (_, value) in
                    let textField = textView as! UITextField
                    view.onChange?([
                        "text": value,
                        "target": view.reactTag,
                        "eventCount": view.nativeEventCount,
                    ])
                }
                let key = reactNode.stringValue
                self.listeners[key] = MaskedRCTBackedTextFieldDelegateAdapter(textField: textView)
                maskedDelegate.listener = self.listeners[key]
                self.masks[key] = maskedDelegate

                textView.delegate = self.masks[key]
            }
        }
    }
```

**Reply by kwonkun** [reply]
+1

**Reply by DorianMazur** [reply]
@arushikesarwani94 Any update on this?

**Reply by DorianMazur** [reply]
@kwonkun @hwhh https://github.com/facebook/react-native/issues/50800#issuecomment-2815120399
